crates/matrix-sdk-crypto/src/store/mod.rs

```diff
@@ -2190,4 +2230,138 @@ mod tests {
 
         assert!(pickle_key.is_err());
     }
+
+    #[async_test]
+    async fn test_build_room_key_bundle() {
```

Comment:
Note to the reviewer: this uses hardcoded `olm::Account` and Megolm session keys, so that I can use a snapshot test to verify the generated bundle.

I like that we can carefully check exactly what is being generated without lots of `assert_eq!` macros, but I don't really like all the boilerplate that goes into setting up the OlmMachines and megolm sessions. What do others think?

(And if we decide to keep it, should `olm_machine_from_account_pickle` and `olm_pickle_data{1,2}` go and live somewhere else?)